app/models/manageiq/providers/openstack/network_manager/network_router.rb

```diff
@@ -30,6 +30,219 @@ class ManageIQ::Providers::Openstack::NetworkManager::NetworkRouter < ::NetworkR
+          }],
+          :isRequired      => true,
+          :options         => ems.cloud_tenants.map do |ct|
+            {
+              :label => ct.name,
+              :value => ct.id.to_s,
+            }
+          end,
+          :includeEmpty    => true,
+          :clearOnUnmount  => true,
+        },
+        {
+          :component => 'sub-form',
+          :title     => _('Router Information'),
+          :id        => 'router_information',
+          :name      => 'router_information',
+          :fields    => [
+            {
+              :component       => 'text-field',
+              :id              => 'router_name',
+              :name            => 'name',
+              :label           => _('Router Name'),
+              :validateOnMount => true,
+              :validate        => [{
+                :type    => 'required',
+                :message => _('Required'),
+              }],
+              :isRequired      => true,
+              :clearOnUnmount  => true,
+            },
+            {
+              :component    => 'switch',
+              :id           => 'admin_state_up',
+              :name         => 'admin_state_up',
+              :label        => _('Administrative State'),
+              :onText       => _('Up'),
+              :offText      => _('Down'),
+              :initialValue => true,
+            },
+          ]
+        },
+        {
+          :component => 'sub-form',
+          :title     => _('External Gateway'),
+          :id        => 'external_gateway',
+          :name      => 'external_gateway',
+          :fields    => [
+            {
+              :component => 'switch',
+              :id        => 'enable',
+              :name      => 'enable',
+              :label     => _('Enable'),
+              :onText    => _('Yes'),
+              :offText   => _('No'),
+            },
+            {
+              :component    => 'switch',
+              :id           => 'source_nat',
+              :name         => 'source_nat',
+              :label        => _('Source NAT'),
+              :onText       => _('Yes'),
+              :offText      => _('No'),
+              :condition    => {
+                :when => 'enable',
+                :is   => true,
+              },
+              :initialValue => true,
+            },
+            {
+              :component      => 'select',
+              :id             => 'network_id',
+              :name           => 'network_id',
+              :key            => "network-#{ems.id}",
+              :label          => _('Network'),
+              :placeholder    => "<#{_('Choose')}>",
+              :includeEmpty   => true,
+              :clearOnUnmount => true,
+              :condition      => {
+                :when => 'enable',
+                :is   => true,
+              },
+              :options        => ems.cloud_networks.select { |cn| cn.ems_id == ems.id && cn.external_facing == true }.map do |cn|
```

Comment:
I'll try these out and push my changes

Reply:
@kbrock `:public_networks` looks like it has the same result but is exactly equivalent to doing 
` :options        => ems.cloud_networks.where(:external_facing => true).map do |cn|` ?

Reply:
It looks like what you want.

Since it goes through routers, I'm not sure that they are exactly the same. but if it gives the same results, then maybe using that makes sense